tests/unit_tests/bittensor_tests/utils/test_network_utils.py

```diff
@@ -110,6 +111,40 @@ def test_upnpc_create_port_map(mocked_upnp):
     with pytest.raises(UPNPCException):
         upnpc_create_port_map(port=port)
     
+class TestGetFormattedWSEndpointURL(unittest.TestCase):
```

Comment:
This kind of test suite is a very nice candidate to use a data provider so we can read in a few lines the different cases  of input/outputs we have here.

Take a look at: https://pypi.org/project/unittest-data-provider/

I am not suggesting that exactly library but the intention itself

Reply:
that's a good idea. I'll modify it to use something similar

Reply:
Great!